File: InAppChat/src/main/java/io/inappchat/inappchat/utils/DisposableList.java
package io.inappchat.inappchat.utils;

import io.reactivex.disposables.Disposable;
import java.util.ArrayList;

public class DisposableList {

  private ArrayList<Disposable> disposables = new ArrayList<>();

  public void add(Disposable d) {
    disposables.add(d);
  }

  public void remove(Disposable d) {
    disposables.remove(d);
  }

  public void dispose() {
    for (Disposable d : disposables) {
      d.dispose();
    }
    disposables.clear();
  }
}

File: InAppChat/src/main/java/io/inappchat/inappchat/utils/DisposableList.kt
package io.inappchat.inappchat.utils

import io.reactivex.rxjava3.disposables.Disposable
import java.util.ArrayList

class DisposableList {
    private val disposables = ArrayList<Disposable>()
    fun add(d: Disposable) {
        disposables.add(d)
    }

    fun remove(d: Disposable) {
        disposables.remove(d)
    }

    fun dispose() {
        for (d in disposables) {
            d.dispose()
        }
        disposables.clear()
    }
}